rt_events.go

```diff
@@ -0,0 +1,144 @@
+package dht
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+func NewRTEvent(
```

Comment:
consider 2 constructors. one for a peer update event, and the other for a bucket refresh event.

The other way to do this would be to have both `RoutingTablePeerUpdateEvent` and `BucketRefreshLaunchedEvent` implement a `RoutingTableEvent` interface - namely that both are json serializable. That would avoid the need of having two nested layers of constructors when used, which is a bit of a smell.

Reply:
> consider 2 constructors. one for a peer update event, and the other for a bucket refresh event.

I'm fine with whatever, I was mostly copying the strategy @petar used for the lookup events in events.go. We should probably change them together if we're going to change them.

Also, unrelated I'm going to rename this function to `NewRoutingTableEvent`